lib/fizzy/execute.cpp

```diff
@@ -18,6 +18,71 @@ struct LabelContext
 };
 
 
+void match_imported_functions(const std::vector<TypeIdx>& module_imported_types,
+    const std::vector<ImportedFunction>& imported_functions)
+{
+    if (module_imported_types.size() != imported_functions.size())
+    {
+        throw std::runtime_error("Module requires " + std::to_string(module_imported_types.size()) +
+                                 " imported functions, " +
+                                 std::to_string(imported_functions.size()) + " provided");
+    }
+}
+
+void match_imported_globals(const std::vector<bool>& module_imports_mutability,
+    const std::vector<ImportedGlobal>& imported_globals)
+{
+    if (module_imports_mutability.size() != imported_globals.size())
+    {
+        throw std::runtime_error(
+            "Module requires " + std::to_string(module_imports_mutability.size()) +
+            " imported globals, " + std::to_string(imported_globals.size()) + " provided");
+    }
+
+    for (size_t i = 0; i < imported_globals.size(); ++i)
+    {
+        if (imported_globals[i].is_mutable != module_imports_mutability[i])
+        {
+            throw std::runtime_error("Global " + std::to_string(i) +
+                                     " mutability doesn't match module's global mutability");
+        }
+        if (imported_globals[i].value == nullptr)
+        {
+            throw std::runtime_error(
+                "Global " + std::to_string(i) + " has a null pointer to value");
+        }
+    }
+}
+
+std::vector<TypeIdx> match_imports(const Module& module,
+    const std::vector<ImportedFunction>& imported_functions,
+    const std::vector<ImportedGlobal>& imported_globals)
+{
+    std::vector<TypeIdx> imported_function_types;
+    std::vector<bool> imported_globals_mutability;
+    for (auto const& import : module.importsec)
+    {
+        switch (import.kind)
+        {
+        case ExternalKind::Function:
```

Comment:
Why are these supported in this PR?

Reply:
This was moved from instantiate to a separate function https://github.com/wasmx/fizzy/pull/77/files#diff-da8232128fc88a65268ab30bf8e8abdcL203

Reply:
I'm not sure that link clears it up for me, it seems to link to the `global_set` case?

Reply:
It links to the line 203 of the old version